### PR TITLE
improve FindProjects

### PR DESCRIPTION
**Pull request by yp05327**
I found `FindAndCount` which can `Find` and `Count` in the same time
Maybe it is better to use it in `FindProjects`

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: models/project/project.go [hunk 1]
```diff
@@ -217,15 +217,10 @@ func CountProjects(ctx context.Context, opts SearchOptions) (int64, error) {
 
 // FindProjects returns a list of all projects that have been created in the repository
 func FindProjects(ctx context.Context, opts SearchOptions) ([]*Project, int64, error) {
-	e := db.GetEngine(ctx)
+	e := db.GetEngine(ctx).Table("project")
```
Comment (wolfogre) [inline models/project/project.go]: IIRC, we don't really need this, `FindAndCount` will detect the table name.

Comment (yp05327) [reply models/project/project.go]: ‘FindAndCount’ is not directly exposed here, so I added ‘Table’. 
It seems that only an xorm session can use ‘FindAndCount’ directly, but db.GetEngine returns ‘Engine’ which is not an xorm session here. Is it possible to get an xorm session here without using ‘Table’?

Comment (wolfogre) [reply models/project/project.go]: Maybe:

```go
func FindProjects(ctx context.Context, opts SearchOptions) ([]*Project, int64, error) {
	e := db.GetEngine(ctx).Where(opts.toConds())
	if opts.Page > 0 {
		e.Asc("created_unix")
	}
	
	projects := make([]*Project, 0, setting.UI.IssuePagingNum)
	count, err := e.FindAndCount(&projects)
	return projects, count, err
}
```

**Review by delvh — APPROVED** [review]
I am fine with both the proposed options as they should be equivalent.

File: models/project/project.go (hunk 1 above)
Comment (yp05327) [reply models/project/project.go]: `db.GetEngine` won't return a session but `Where` will. I missed this. Thanks a lot.

**Pull request merged**